jvb/src/main/kotlin/org/jitsi/videobridge/cc/allocation/BitrateController.kt

```diff
@@ -278,14 +276,11 @@ class BitrateController<T : MediaSourceContainer> @JvmOverloads constructor(
             packetHandler.allocationChanged(allocation)
 
             if (useSourceNames) {
-                // If rewriting SSRCs, all active sources will be notified separately.
-                if (!doSsrcRewriting) {
-                    // TODO as per George's comment above: should this message be sent on message transport connect?
-                    val newForwardedSources = allocation.forwardedSources
-                    if (forwardedSources != newForwardedSources) {
-                        forwardedSources = newForwardedSources
-                        eventEmitter.fireEvent { forwardedSourcesChanged(newForwardedSources) }
-                    }
+                // TODO as per George's comment above: should this message be sent on message transport connect?
```

Comment:
This part now runs when doing ssrc rewriting - I don't understand this?

Reply:
SSRC-rewriting used to work on top of the "source list" which changes when the ordered of endpoints change. It was supressing this message and sending its own version that contained all mapped sources (if you look at the Endpoint.kt diff I removed a "sendForwardedSourcesMessage" line).

Now it works on top of effective constraints (so it respects the receiver's preference), and it's independent of the bandwidth allocation code. The bandwidth allocation code does its thing and sends a list of sources to the receiver ("forwarded sources"). Independently, ssrc-rewriting sends mappings. 

"Effective constraints" takes into accoount all of various last-n limits, ssrc-limits, receiver preferences and video availability. Bandwidth allocation only forwards sources allowed by effective constraints. SSRC-rewriting only maps sources enabled in effective constraints.